packages/docusaurus-theme-common/src/utils/__tests__/docsUtils.test.tsx

```diff
@@ -762,3 +763,33 @@ describe('useCurrentSidebarCategory', () => {
     );
   });
 });
+
+describe('filterDocCardListItems', () => {
+  it('returns original list of valid items', () => {
+    const items = [
+      {type: 'link', href: '/foo/bar', label: 'Foo'},
+      {type: 'link', href: 'bar/foo', label: 'Bar'},
+      {type: 'link', href: '/foo/listed', label: 'Listed'},
+    ];
+    expect(filterDocCardListItems(items)).toEqual(items);
+  });
```

Comment:
maybe add html items here, and also valid categories with valid items as children